lib/puppet/file_bucket/dipper.rb

```diff
@@ -56,6 +58,46 @@ def backup(file)
     end
   end
 
+  # Diffs two filebucket files identified by their sums
+  def diff(suma, sumb, filea, fileb)
```

Comment:
Nitpick: the locals being passed to this method I think weren't as confusing, named `checksum_a`, `checksum_b`, `file_a` and `file_b` respectively.  `sum_a` and `sum_b` is also fine, but I think we should make it slightly more readable with an underscore.